tools/data_builder.py

```diff
@@ -113,19 +166,38 @@ def parse_state(state, text):
             entry["date"] = date
             entry["date_text"] = date_text
         elif starts_with == '*':
-            link = url_regex.search(line)
-            if link:
-                entry["links"].append(link.group())
+            link_text, link_url = find_md_link_or_url(line)
+            if link_url:
+                entry["links"].append(link_url)
+                entry["links_v2"].append({
+                    "url": link_url,
+                    "text": link_text,
+                })
             else:

```

Comment:
shouldn't this fail the CI pipeline?  Otherwise it could be hard to keep track of failed links

Reply:
Good point. I wasn't thinking about failure modes much. I initially wanted the data build to always work. But now that it runs on every PR, we would notice when it fails. An `exit(1)` will let people know the data is broken.